dashboard/test/helpers/teacher_application_helper_test.rb

```diff
@@ -38,22 +44,46 @@ def sign_in(user)
       {
         expected_output: true,
         condition_message: 'application exists and is incomplete',
-        user: @applicant_with_incomplete_app
+        user: @user_with_two_incomplete_apps
       },
       {
         expected_output: false,
-        condition_message: 'application exists and is unreviewed',
-        user: @applicant_with_unreviewed_app
+        condition_message: 'application exists and is reopened',
+        user: @user_with_reopened_app
       },
       {
         expected_output: false,
         condition_message: 'application is in a different year',
-        user: @teacher_with_not_current_app
+        user: @user_with_outdated_incomplete
       }
     ].each do |expected_output:, user:, condition_message:|
       sign_in user
       assert has_incomplete_application?, "expected true when #{condition_message}" if expected_output
       refute has_incomplete_application?, "expected false when #{condition_message}" unless expected_output
     end
   end
+
+  test "has_reopened_application" do
+    [
+      {
+        expected_output: true,
+        condition_message: 'application exists and is reopened',
+        user: @user_with_reopened_app
+      },
+      {
+        expected_output: false,
+        condition_message: 'application exists and is incomplete',
+        user: @user_with_two_incomplete_apps
+      },
+      {
+        expected_output: false,
+        condition_message: 'application is in a different year',
+        user: @user_with_outdated_reopened
+      }
+    ].each do |expected_output:, user:, condition_message:|
+      sign_in user
+      assert has_reopened_application?, "expected true when #{condition_message}" if expected_output
+      refute has_reopened_application?, "expected false when #{condition_message}" unless expected_output
```

Comment:
I think this will do what you want a bit more concisely:
```suggestion
      assert_equal expected_output, has_reopened_application?, condition_message
```

Reply:
Perhaps a different refactor needs to happen. `has_reopened_application?` will return `nil` when the user doesn't have a current application, and `false` when the user has a current application but it's not the status `reopened`. Is that bad form? Then the `assert_equal` gives the warning that the check should be `assert_nil`

Reply:
Wait, lies, it doesn't give an error, but the test fails

Reply:
a good practice would be for a boolean method like this one to return true or false. if you don't want to change that now, you can use `!! has_reopened_application?` here instead.

Reply:
Ah, got it. Is there a good Ruby way to force `current_application && current_application.status == 'incomplete'` to return false if there's no `current_application`?

Reply:
Yeah, the `current_application` is the part that can return nil, so it would be
```
 !!current_application && current_application.status == 'incomplete'
```

Reply:
Ooo clever, thanks!